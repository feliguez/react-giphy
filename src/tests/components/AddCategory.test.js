import React from 'react';
import { shallow } from 'enzyme';
import AddCategory from '../../components/AddCategory';

describe('Test <AddCategory />', () => {
  const setCategories = jest.fn();
  let wrapper = shallow(<AddCategory setCategories={setCategories} />);

  beforeEach(() => {
    jest.clearAllMocks();
    wrapper = shallow(<AddCategory setCategories={setCategories} />);
  });

  test('should be equal to snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
  test('should change input text', () => {
    const input = wrapper.find('input');
    const value = 'hola mundo';
    input.simulate('change', { target: { value } });
    // console.log(input.props());

    expect(wrapper.find('p').text().trim()).toBe(value);
  });
  test('should NOT submit data', () => {
    wrapper.find('form').simulate('submit', { preventDefault: () => {} });
    // console.log(input.props());

    expect(setCategories).not.toHaveBeenCalled();
  });
  test('should call setCategories and clean input text', () => {
    const value = 'hola mundo';

    wrapper.find('input').simulate('change', { target: { value } });
    // console.log(wrapper.find('input').props());
    wrapper.find('form').simulate('submit', { preventDefault: () => {} });
    // console.log(wrapper.find('input').props());

    expect(setCategories).toHaveBeenCalled();
    expect(setCategories).toHaveBeenCalledTimes(1);
    expect(setCategories).toHaveBeenCalledWith(expect.any(Function));
    expect(wrapper.find('input').props().value).toBe('');
  });
});
