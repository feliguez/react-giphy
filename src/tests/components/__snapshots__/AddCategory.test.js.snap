// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <AddCategory /> should be equal to snapshot 1`] = `
<form
  onSubmit={[Function]}
>
  <p />
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
